tests/components/auth/test_init.py

```diff
@@ -519,6 +519,62 @@ async def test_ws_delete_refresh_token(
     assert refresh_token is None
 
 
+@pytest.mark.parametrize("raise_error", [False, True])
+async def test_ws_delete_all_refresh_tokens(
+    hass: HomeAssistant,
+    hass_admin_user: MockUser,
+    hass_admin_credential: Credentials,
+    hass_ws_client: WebSocketGenerator,
+    hass_access_token: str,
+    raise_error: bool,
+) -> None:
+    """Test deleting all refresh tokens."""
+    assert await async_setup_component(hass, "auth", {"http": {}})
+
+    # one token already exists
+    await hass.auth.async_create_refresh_token(
+        hass_admin_user, CLIENT_ID, credential=hass_admin_credential
+    )
+    token = await hass.auth.async_create_refresh_token(
+        hass_admin_user, CLIENT_ID + "_1", credential=hass_admin_credential
+    )
+
+    if raise_error:
```

Comment:
Conditional logic makes tests hard to read and fragile. I suggest rewriting the tests to avoid this. Make two separate tests if needed.

Reply:
Tests are splitted